File: ClubTableTracker.Server/Models/Club.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
namespace ClubTableTracker.Server.Models;

public class Club
{
    public int Id { get; set; }
    [MaxLength(100)]  public string Name        { get; set; } = string.Empty;
    [MaxLength(1000)] public string Description { get; set; } = string.Empty;
    [MaxLength(64)]   public string AccessKey   { get; set; } = string.Empty;
    public string OpenTime { get; set; } = "10:00";
    public string CloseTime { get; set; } = "22:00";
    public string? LogoUrl { get; set; }
    public List<GameTable> Tables { get; set; } = new();
    public List<ClubMembership> Memberships { get; set; } = new();
}

File: MobileFlutter/lib/models/club.dart
class Club {
  final int id;
  final String name;
  final String description;
  final String openTime;
  final String closeTime;
  final String? logoUrl;

  const Club({
    required this.id,
    required this.name,
    required this.description,
    required this.openTime,
    required this.closeTime,
    this.logoUrl,
  });

  factory Club.fromJson(Map<String, dynamic> json) => Club(
        id: json['id'] as int,
        name: json['name'] as String,
        description: json['description'] as String? ?? '',
        openTime: json['openTime'] as String? ?? '09:00',
        closeTime: json['closeTime'] as String? ?? '22:00',
        logoUrl: json['logoUrl'] as String?,
      );
}
